app/controllers/profiles_controller.rb

```diff
@@ -14,9 +14,19 @@ def update
     end
   end
 
+  def update_email
+    if resource.update_with_password(update_params)
+      sign_out(resource)
```

Comment:
Here it's unavoidable I guess because of our localStorage also need to be updated with latest email etc and so the contexts. So before updating the email, in the UI make sure to show an alert that this will sign them out.